asteroid/engine/solver.py

```diff
@@ -95,69 +110,92 @@ def train(self):
                 self.val_no_impv = 0
                 self.early_val_no_impv = 0
             # Adjust learning rate (halving)
-            if self.half_lr and self.val_no_impv >= 3:
+            if self.tr_conf['half_lr'] and self.val_no_impv >= 3:
                 self.multiply_learning_rate(self.optimizer, 0.5)
                 self.ep_half_lr.append(epoch)
                 self.val_no_impv = 0
             # Early stopping
-            if self.early_val_no_impv >= 8 and self.early_stop:
+            if self.early_val_no_impv >= 8 and self.tr_conf['early_stop']:
```

Comment:
Remove constants. Put it in default dictionary.

Reply:
Same for 1.001 in line 106

Reply:
You're right, thanks. Callbacks will make this so much better